vmdb/app/models/miq_event_catcher_openstack_infra.rb

```diff
@@ -0,0 +1,18 @@
+class MiqEventCatcherOpenstackInfra < MiqEventCatcher
```

Comment:
If EventCatcherOpenstackInfra is a subclass of EventCatcherOpenstack, then the models should do the same.  In fact, I thikn the method all_valid_ems_in_zone is identical at that point, right?  So, then you wouldn't need that method.